### DISCUSSION TITLE
Mojolicious Lite websockets over TLS

### DISCUSSION THREAD

**Discussion by squizzster** [mojolicious/mojo#1751] [Q&A] [answered]
Hi there and what great software Mojo is.

I have a simple app which looks like:

`
websocket '/gws/*' => sub {
   message::process_websocket_request(shift);
};

get '*' => sub {
  my $c = shift;
  process_request($c);
  return 1;
};
`
And it works - it's super amazing just how easy it was.

However, it only works with ws: and does not work with **wss://**
With wss:// the "get '*'" routine is processing the request and not the web-socket routine.
The request does reach the get "*".

I would like both ws: and wss: to be processed by the process_websocket routine.

I do proxy https requests using APACHE, I wonder if this could be the issue....

Thanks for any help, 
I'll keep trying to figure it out!

Mark

**Comment by Grinnz** [comment, marked as answer]
mod_proxy_wstunnel is required to proxy websockets through Apache: https://docs.mojolicious.org/Mojolicious/Guides/Cookbook#Apache-mod_proxy

Also `get '*'` is wrong, should be `get '/*'`

**Reply by squizzster** [reply]
Hey, wow, that was fast and thank you... I thought it might be something like that. 
I will fix the get.
Good man and thank you again.
